sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/ResolveDefaultColumnsUtil.scala

```diff
@@ -344,3 +358,13 @@ object ResolveDefaultColumns {
     }
   }
 }
+
+/**
+ * Represents a table with an underlying raw schema to use for resolving DEFAULT column references.
+ * The 'ignoreColumnsForInserts' field represents a set of column names to exclude from
+ * consideration when performing INSERT INTO commands, if any.
+ */
+trait TableWithRawSchemaForColumnDefaults {
+  def rawSchema: Option[StructType]
```

Comment:
Why the result is optional? Could you write an example about the difference between `rawSchema` and `schema`?

Reply:
Good point, I changed this to become non-optional an added a comment with an example of how this might differ from the regular schema.